### PR TITLE
[docs] Add component context to contributing guide

### PR DESCRIPTION
**Pull request by hickeyma**
Adds component context including references to tutorial.

This PR improves the usability around the process of creating a component for new users.

Partial https://github.com/open-telemetry/opentelemetry-collector/issues/5675

### CODE REVIEW

**Comment by hickeyma** [comment]
No changelog required.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -75,10 +75,14 @@ providing the following information:
...
+Components comprise of exporters, extensions, receivers, and processors. The key criteria to implementing a component is to:
 
-Any component (receiver, processor, exporter, or extension) needs to implement the interfaces defined on the [core
-repository](https://github.com/open-telemetry/opentelemetry-collector). Familiarize yourself with the interface of the
-component that you want to write, and use existing implementations as reference.
+* Implement the `component.Component` interface
+* Provide a configuration structure which defines the configuration of the component
+* Provide the implementation which performs the component operation
+
+Familiarize yourself with the interface of the
```
Comment (fatsheep9146) [inline CONTRIBUTING.md:84]: There should not be a newline?

Comment (hickeyma) [reply CONTRIBUTING.md]: Removed the newline

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -75,10 +75,14 @@ providing the following information:
 * The configuration options your component will accept. This will help us understand what it does and have an idea of
   how the implementation might look like.
 
+Components comprise of exporters, extensions, receivers, and processors. The key criteria to implementing a component is to:
 
-Any component (receiver, processor, exporter, or extension) needs to implement the interfaces defined on the [core
-repository](https://github.com/open-telemetry/opentelemetry-collector). Familiarize yourself with the interface of the
-component that you want to write, and use existing implementations as reference.
+* Implement the `component.Component` interface
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply CONTRIBUTING.md]: Added the link to Go doc

**Comment by hickeyma** [comment]
@fatsheep9146 @mx-psi Thank you for the review. I have updated the comments and it is ready for review again.

**Review by mx-psi — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**